src/qililab/digital/circuit_transpiler.py

```diff
@@ -14,20 +14,26 @@
 
 """Circuit Transpiler class"""
 
+from typing import TYPE_CHECKING, Union
+
 import networkx as nx
 from qibo.models import Circuit
-from qibo.transpiler.placer import Placer
-from qibo.transpiler.router import Router
 
 from qililab.config import logger
 from qililab.digital.circuit_optimizer import CircuitOptimizer
 from qililab.digital.circuit_router import CircuitRouter
 from qililab.digital.circuit_to_pulses import CircuitToPulses
-from qililab.pulse.pulse_schedule import PulseSchedule
 from qililab.settings.digital.digital_compilation_settings import DigitalCompilationSettings
 
 from .gate_decompositions import translate_gates
 
+if TYPE_CHECKING:
+    from qibo.transpiler.placer import Placer
+    from qibo.transpiler.router import Router
+
+    from qililab.platform.platform import Platform
+    from qililab.pulse.pulse_schedule import PulseSchedule
```

Comment:
nice!

![the-mandalorian-this-is-the-way](https://github.com/user-attachments/assets/6e47ff23-8b89-4de6-a900-0fd0640fb9ea)